### DISCUSSION TITLE
Do blocks of a HA Prometheus setup get uploaded 2 times to the object storage by the 2 sidecars?

### DISCUSSION THREAD

**Discussion by amine250** [thanos-io/thanos#4474] [Questions & Answers] [answered]
Hi, I have a high availability prometheus setup with two {prometheus+thanos sidecar}.

Correct me if I'm wrong, what I understood is that the 2 sidecars will both upload the same metrics to the object storage.

Basically, to have a setup with HA, the price is double network io (upload and download) and cpu for deduplication.

I also wonder how does the compactor behave in this case, one could imagine that the compactor could also deduplicate blocks, and compacts two into one, the same way the querier does.

Thoughts?

**Comment by yeya24** [comment, marked as answer]
> I also wonder how does the compactor behave in this case, one could imagine that the compactor could also deduplicate blocks, and compacts two into one, the same way the querier does.

This is already supported in the latest v0.22.0 release. Compactor will do the same thing as querier to deduplicate the two blocks and merge them into one.

Please see https://thanos.io/v0.22/components/compact.md/#vertical-compaction-use-cases.
